### ISSUE TITLE
Nested validation display errors 

### ISSUE THREAD

**Issue by stanislavdev1993** [yiisoft/validator#147]
```php
$request = [
    'author' => [
        'name' => '',
        'age' => '',
    ],
];
 
$rule = Nested::rule([
    'age' => [
        Number::rule(),
    ],
    'name' => [
        HasLength::rule()
    ],
]);
```
 
Got errors

```php
$errors = ['author': ['Value cannot be blank.', 'Value cannot be blank.']]
```
 
Better way to show in what nested attribute got errors

```php
$errors = ['author': ['name': ['Value cannot be blank.'], 'age': ['Value cannot be blank.']]]
```

**Comment by xepozz** [comment]
I think that proposed solution can be improved due to the possibility of unexpected result format.
I have modified your snippet and made an example:
```php
$request = [
    'id' => 1,
    'post' => [
        'author' => [
            'identity' => [
                'name' => '',
                'age' => '',
            ]
        ],
    ],
];
 
$rule = Nested::rule([
    'author.identity.age' => [
        Number()->required(),
    ],
    'author.identity.name' => [
        HasLength::rule()->required(),
    ],
]);
```
The errors will be uncontrollable tree
```php
$result = [
    'post' => [
        'author' => [
            'identity' => [
                'name' => ['Value cannot be blank.'], 
                'age' => ['Value cannot be blank.'],
            ],
        ],
    ],
];
```

One way how to get the error messages is to check that array keys are in int format.

----

If we concatenate nested attributes with `.` symbol, then we need to escape or replace that symbol or even make it reservable.

In conclusion I want to say that we need to think about other formats and choose the best one before implementing.

**Comment by arogachev** [comment]
@stanislavdev1993 Where you got this from?

```php
Number()->required(),
```

It's not in API. Or it's sort of draft / pseudocode?

```php
$errors = ['author': ['Value cannot be blank.', 'Value cannot be blank.']]
```

For me output is flat array. Could you show the rest of the code?

**Comment by stanislavdev1993** [comment]
@arogachev It was mistake. I fixed this in comment.

**Comment by arogachev** [comment]
Got it, no problem. But what about chained `required` call?

**Comment by stanislavdev1993** [comment]
@arogachev Done. It just a sample. issue with errors diplaying in nested rule.
